File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        curr_max=[0]
        def dfs(root):
            if not root:
                return 0
            if not root.left and not root.right:
                return 1
            l=dfs(root.left)
            r=dfs(root.right)
            curr_max[0]=max(curr_max[0],l+r)
            return max(l,r)+1
        dfs(root)
        return curr_max[0]

File: Data Structures & Algorithms/sum-root-to-leaf-numbers/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans=0;
    public int sumNumbers(TreeNode root) {
        int t=0;
        helper(root,t);
        return ans;
    }
    public void helper(TreeNode root, int t){
        if(root==null) return;
        t=(t*10)+root.val;
        if(root.left==null && root.right==null){
            ans+=t;
            return;
        }
        helper(root.left,t);
        helper(root.right,t);
    
    }
}